app/uk/gov/hmrc/customerprofile/controllers/GetApplePassController.scala

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.customerprofile.controllers
+
+import com.google.inject.{Inject, Singleton}
+import play.api.Logger
+
+import javax.inject.Named
+import play.api.libs.json.Json
+import play.api.libs.json.Json.{obj, toJson}
+import play.api.mvc.{Action, ActionBuilder, AnyContent, BodyParser, ControllerComponents, Request, Result}
+import uk.gov.hmrc.api.controllers._
+import uk.gov.hmrc.auth.core.AuthorisationException
+import uk.gov.hmrc.customerprofile.auth._
+import uk.gov.hmrc.customerprofile.connector.ShutteringConnector
+import uk.gov.hmrc.customerprofile.domain.types.ModelTypes.JourneyId
+import uk.gov.hmrc.customerprofile.services.GetApplePassService
+import uk.gov.hmrc.http.{HeaderCarrier, NotFoundException, Upstream4xxResponse}
+import uk.gov.hmrc.play.bootstrap.backend.controller.BackendController
+import uk.gov.hmrc.play.http.HeaderCarrierConverter.fromRequest
+import uk.gov.hmrc.customerprofile.domain._
+import uk.gov.hmrc.domain.Nino
+
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class GetApplePassController @Inject()(
+  service:                                          GetApplePassService,
+  accessControl:                                    AccountAccessControl,
+  @Named("citizen-details.enabled")   val citizenDetailsEnabled: Boolean,
+  controllerComponents:                             ControllerComponents,
+  shutteringConnector:                              ShutteringConnector
+)(implicit val executionContext                     :ExecutionContext)
+    extends BackendController(controllerComponents)
+    with HeaderValidator {
+  outer =>
+  val logger: Logger = Logger(this.getClass)
+   def parser: BodyParser[AnyContent] = controllerComponents.parsers.anyContent
+
+  private final val WebServerIsDown = new Status(521)
+  val app                          = "Apple Pass"
+
+  def invokeAuthBlock[A](
```

Comment:
It might be cleaner to extract all the authorisation logic into an `Authorisation` trait